solver/src/driver/solver_settlements.rs

```diff
@@ -44,28 +55,16 @@ pub fn merge_settlements(
     name: &'static str,
     settlements: Vec<Settlement>,
 ) -> SolverWithSettlements {
-    let mut settlements = settlements
-        .into_iter()
-        .map(|settlement| {
-            let objective_value = settlement.objective_value(prices);
-            RatedSettlement {
-                settlement,
-                objective_value,
-            }
-        })
-        .collect::<Vec<_>>();
-    settlements.sort_by(|a, b| b.objective_value.cmp(&a.objective_value));
-    if let Some(settlement) = merge_at_most_settlements(
-        max_merged_settlements,
-        settlements
-            .iter()
-            .map(|settlement| settlement.settlement.clone()),
-    ) {
-        let objective_value = settlement.objective_value(prices);
-        settlements.push(RatedSettlement {
-            settlement,
-            objective_value,
-        });
+    let mut settlements = settlements;
```

Comment:
You can change it to `mut settlements: Vec<Settlement>,` in the function arguments and remove this line.